modules/amr/amr.c

```diff
@@ -248,7 +292,15 @@ static int decode_wb(struct audec_state *st,
 	if (fmt != AUFMT_S16LE)
 		return ENOTSUP;
 
-	IF2D_IF_decode(st->dec, &buf[1], sampv, 0);
+	if (amr_ac->aligned) {
+		IF2D_IF_decode(st->dec, &buf[1], sampv, 0);
+	}
+	else {
+		uint8_t temp[len];
```

Comment:
we should try to avoid variable-length arrays, this is considered to be insecure coding.

Reply:
Maybe we can add this with `-Wvla` as compiler warning flag? I found only one other issue:

```
modules/gst/gst.c:160:13: warning: variable length array used [-Wvla]
        int16_t buf[st->sampc];
```

Reply:
I think this is a great idea :)

thanks for fixing the gst module

Reply:
The max size of len is already protected by checks for size NB_SERIAL_MAX. See rows 289 and 356. 

Should I commit change where these temp arrays are always allocated at the max allowed size, i.e. NB_SERIAL_MAX and +1 ? 

thanks

Reply:
it would better to declare a fixed-size buffer that is big enough to hold a frame.

the buffer can be located either on the stack, or in the instance.

Reply:
see new commit using fixed-size buffer. tested ok.